### DISCUSSION TITLE
Unable to trigger BeforeProcessPipeline and AfterProcessPipeline

### DISCUSSION THREAD

**Discussion by ShivamSinghal9** [redis/go-redis#2189] [General]
Over the "Hook" interface, we have well-defined methods with their definition called BeforeProcess, AfterProcess, BeforeProcessPipeline, and AfterProcessPipeline. I am able to activate `BeforeProcess` and `AfterProcess`, but i cannot trigger `BeforeProcessPipeline` and `AfterProcessPipeline`. Is there a specific reason for that?

**Comment by vmihailenco** [comment]
Hard to tell without seeing the code. Did you try to execute a [pipeline](https://redis.uptrace.dev/guide/go-redis-pipelines.html)?
